suzieq/poller/worker/nodes/node.py

```diff
@@ -479,205 +383,215 @@ async def _parse_device_type_hostname(self, output, _) -> None:
                 if hmatch:
                     hostname = hmatch.group(1)
             elif devtype == "nxos":
-                data = output[3]["data"]
-                hostname = data.strip()
+                hgrp = re.search(r'[Dd]evice\s+name:\s+(\S+)', data)
+                if hgrp:
+                    hostname = hgrp.group(1)
+                else:
+                    hostname = self.address
             elif devtype == "eos":
-                data = output[3]['data']
-                lines = data.split('\n')
-                hostname = lines[1].split('FQDN:')[1].strip()
+                # We'll fill in the hostname when the node gets re-init
+                hostname = None
             elif devtype in ["iosxe", "ios"]:
                 matchval = re.search(r'(\S+)\s+uptime', output[0]['data'])
                 if matchval:
                     hostname = matchval.group(1).strip()
                 else:
                     hostname = self.address
-            elif devtype != "iosxr" and output[3]["status"] == 0:
-                hostname = output[3]["data"].strip()
+            elif devtype != "iosxr":
+                hgrp = re.search(r'(\S+)\s+uptime\s+is', data)
+                if hgrp:
+                    hostname = hgrp.group(1)
+                else:
+                    hostname = self.address
 
-        elif output[2]["status"] == 0:
-            data = output[2]["data"]
+        elif output[1]["status"] == 0:
+            data = output[1]["data"]
             if "Cumulus Linux" in data:
                 devtype = "cumulus"
             else:
                 devtype = "linux"
 
             version_str = data
-            if output[1]['status'] == 0:
-                # Hostname is the only line of /bin/hostname
-                hostname = output[1]['data'].strip()
-
-            self._extract_nos_version(data)
+            # Hostname is the last line of the output
+            hostname = data.splitlines()[-1].strip()
```

Comment:
check data.splitlines() before accessing to [-1]

Reply:
Fixed.